_posts/2020-03-11-how-to-avoid-null-reference-exception.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: post
+title: "How to avoid NullReferenceException in C#"
+date: 2020-03-11 03:57:07
+tags:
+- .NET
+- C#
+categories:
+- development
+twitter_text: How to avoid NullReferenceException in C#
+authors: Cheng Yang
+image: https://images.unsplash.com/photo-1510915228340-29c85a43dcfe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80
+image_url: https://unsplash.com/photos/9SoCnyQmkzI
+image_credit: Jefferson Santos
+---
+
+I have been working as a software developer for almost three years, the most common exception or bug I made is [NullReferenceException](https://docs.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=netframework-4.8) -`System.NullReferenceException: Object reference not set to an instance of an object.` This exception is thrown when you try to access any properties / methods/ indexes on a type of object which points to null. 
+
+## Common Scenario 1:
+
+```csharp
+using System; 
+public class Program
+{
+    public static void Main()
+    {
+     	 string dog = null;
+		   var value = dog.ToString(); //Object reference not set to an instance of an object
```

Comment:
This should line everything up:

```suggestion
		     var value = dog.ToString(); //Object reference not set to an instance of an object
```

Reply:
weird,  the format looks fine in visual studio code
![image](https://user-images.githubusercontent.com/27984559/76520504-62b27f00-6439-11ea-9d1b-6cda4bf18cd2.png)

Reply:
Weird

Reply:
I will update the format if it still looks weird in the blog post.